README typo, CBClient Bump, TROUBLESHOOTING doc (#4)
CloudBoltSoftware/terraform-provider-cloudbolt: TROUBLESHOOTING.md

# Troubleshooting

You may find that the Terraform Provider doesn't always provide the most useful error messages.
Here are some common errors and their workarounds.

## unexpected state 'CART'

```text
$ terraform destroy
... snip ...
cloudbolt_bp_instance.mycbresource: Destroying... [id=/api/v2/resources/service/35/]
cloudbolt_bp_instance.mycbresource: Still destroying... [id=/api/v2/resources/service/35/, 10s elapsed]

Error: Error waiting for Job (/api/v2/orders/81/) to complete: unexpected state 'CART', wanted target 'SUCCESS'. last error: %!s(<nil>)
```

> This means that the "Delete" resource action requires approval in CloudBolt.

Reviewer: Should "resource action" be title case?

Reviewer: Some of these are questions for @jevanscb

Author: ```suggestion
This means that the "Delete" resource action Requires Approval in CloudBolt.
```

Author: I think we should. Makes it more obvious that it's a CloudBolt property and not some other thing.

To resolve this...

> 1. Go to `https://your.cloudbolt.instance/actions/resource_actions/`

Reviewer: I have seen people read URLs like this literally. Would the following make this more obvious?

```
1. Go to `https://<your-cloudbolt-instance>/actions/resource_actions/`
```

Author: ```suggestion
1. Go to `https://<your-cloudbolt-instance>/actions/resource_actions/`
```

2. Edit the "Delete" action.

> 3. De-select "Requires Aproval".

Reviewer 2: ```suggestion
3. De-select "Requires Approval".
```

Reviewer 2: It's unclear if this is just a limitation (can't use the cloudbolt approval process?) or if it's just a recommendation - could a user still utilize the approval process? If so, should we link them to the docs for that process?

Author: The approval process doesn't really fit in the workflow of Terraform since when you execute Terraform it's going to wait for any provisioned resources to be created. If your Terraform script needs to wait for a person to click "Approve" then it'll be waiting a _very_ long time.

That said we could link to some documentation about custom approval processes. For example, if a Terraform user is making a request in a Terraform Group for some specific Blueprints, auto-approve, otherwise require some number of approvals.

> You may also want to login to CloudBolt as the user Terraform is running as and clear your Cart.

Reviewer: I think "login" is a noun

```suggestion
You may also want to log in to CloudBolt as the user Terraform is running as and clear your Cart.
```

## Parameter named 'some_input_name' does not exist

```text
$ terraform apply
... snip ...
cloudbolt_bp_instance.mycbresource: Creating...

Error: received a server error: {"status_code":500,"error":"Parameter named 'some_input_name' does not exist","detail":"An unexpected error occurred."}
```

This means that a Parameter name used in the Terraform Plan does not match what is in CloudBolt.

This can happen for a few reasons, but most often it is a result of a Plugin's generated parameter name differing from the expected name.

> 1. Navigate to `https://your.cloudbolt.instance/actions/{action_id}/`.

Reviewer: Same question as above re: URL

Author: ```suggestion
1. Navigate to `https://<your-cloudbolt-instance>/actions/{action_id}/`.
```
